components/script/textinput.rs

```diff
@@ -442,12 +442,24 @@ impl<T: ClipboardProvider> TextInput<T> {
         let target_line: isize = self.edit_point.line as isize + adjust;
 
         if target_line < 0 {
-            self.edit_point.index = 0;
-            self.edit_point.line = 0;
+            let zero_point = TextPoint { index: 0, line: 0 };
+            self.edit_point = zero_point.clone();
+            match (self.selection_origin, self.selection_direction) {
+                (Some(_), SelectionDirection::None) | (Some(_), SelectionDirection::Forward) => {
+                    self.selection_origin = Some(zero_point)
+                },
+                _ => (),
```

Comment:
Can make use of the if let shorthand here.

Reply:
Ok, I changed to `if`